### PR TITLE
It is now possible to create normal and group comments

### PR DESCRIPTION
**Pull request by eanker**
Fixes preview-code/rite-evaluation#93

runs together with preview-code/rite-evaluation#221

### CODE REVIEW

File: src/main/java/me/previewcode/backend/PullRequestAPI.java [hunk 1]
```diff
@@ -19,12 +17,22 @@
...
+            StatusBody data) {
         owner = owner.toLowerCase();
         name = name.toLowerCase();
         DatabaseReference projectRef = this.ref.child("pulls");
         projectRef.child(owner).child(name).child("branch").child(number)
                 .child("status").setValue(data.status);
         return data;
     }
+
+    public void setComments(String owner, String name, int number,
```
Comment (TimvdLippe) [inline src/main/java/me/previewcode/backend/PullRequestAPI.java]: This is not an end-point, so this method should be placed into a service or separate file.

**Comment by eanker** [comment]
The whole structure is being changed, because it was very cost effective.
Only Thomas is changing the whole structure of the files, therefore this pr is on-hold for the moment.

**Comment by txsmith** [comment]
What's the status of this PR?

File: src/main/java/me/previewcode/backend/DTO/PRGroupComment.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package me.previewcode.backend.DTO;
+
+public class PRGroupComment extends PRComment {
+    public String title;
```
Comment (txsmith) [inline src/main/java/me/previewcode/backend/DTO/PRGroupComment.java]: Why is this called `title` while the PRComment string is called `body`?

File: src/main/java/me/previewcode/backend/DTO/User.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package me.previewcode.backend.DTO;
+
+import java.net.URL;
+
+public class User {
+    public String login;
+    public URL html_url;
+    public String avatar_url;
```
Comment (txsmith) [inline src/main/java/me/previewcode/backend/DTO/User.java]: If `html_url` has type `URL`, then my intuition would be that `avatar_url` should have the same type. Is there a reason why this is a `String`?

File: src/main/java/me/previewcode/backend/GithubAPI.java [hunk 4]
```diff
@@ -8,37 +8,40 @@
...
 
-@Path("/project/{owner}/{name}")
+@Path("{owner}/{name}/pulls/")
 public class GithubAPI extends GithubConnection {
     GHRepository repo;
 
-    public GithubAPI() throws IOException {
+    public GithubAPI() {
         super();
     }
```
Comment (txsmith) [inline src/main/java/me/previewcode/backend/GithubAPI.java]: This constructor seems to be redundant. If `GithubAPI` has no constructor, Java will use the first constructor in the list of super classes. So only calling `super();` in a constructor is equivalent to having no constructor.

File: src/main/java/me/previewcode/backend/GithubAPI.java [hunk 5]
```diff
@@ -8,37 +8,40 @@
...
 
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
-    @Path("/createPR/{head}/{base}")
     public PrNumber createPR(@PathParam("owner") String owner,
-            @PathParam("name") String name, @PathParam("head") String head,
-            @PathParam("base") String base, PRbody body) throws IOException {
+            @PathParam("name") String name, PRbody body) throws IOException {
         PrNumber number = new PrNumber();
+        PullRequestAPI prAPI = new PullRequestAPI();
```
Comment (txsmith) [inline src/main/java/me/previewcode/backend/GithubAPI.java]: Manually calling an endpoint inside another endpoint is a design flaw. `PullRequestAPI` should only be used when the frontend makes a call to it. If `PullRequestAPI` contains functionality that is used in multiple places, it should be extracted away from the endpoint into it's own class and injected using Guice. I suggest we fix this later given the limited time left.

File: src/main/java/me/previewcode/backend/PullRequestAPI.java [hunk 6]
```diff
@@ -19,12 +17,22 @@
     @Path("/{owner}/{name}/{branch}/status/")
     public StatusBody getStatus(@PathParam("owner") String owner,
```
Comment (txsmith) [inline src/main/java/me/previewcode/backend/PullRequestAPI.java]: This has two problems:
 - the naming is deceiving. It *sets* the PR status and returns the new status, `getStatus` is not a very good name.
 - The endpoint is exposed for anyone to call. This means that any user can set the status of a PR, regardless of whether they're authorized to do so.

Solution for this second point would be to refactor this class to not be an endpoint but a service (like @TimvdLippe mentions below), and create a separate endpoint that checks authorization via github and then calls this service.

**Comment by eanker** [comment]
@txsmith On the frontend I had some difficulties regarding the calling of the id.
But I think both pr's will soon be ready to get a new commit :)

**Comment by txsmith** [comment]
Very well! I've extracted #240 for the refactorings. For this PR, I suggest you fix the naming problems and leave the rest for a later fix.

**Comment by eanker** [comment]
Will do

File: src/main/java/me/previewcode/backend/DTO/User.java (hunk 3 above)
Comment (eanker) [inline src/main/java/me/previewcode/backend/DTO/User.java]: We need to fix the framework for this.

**Comment by TimvdLippe** [comment]
LGTM

**Pull request merged**